t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -1280,6 +1280,15 @@ test_expect_success 'sparse index is not expanded: read-tree' '
 	ensure_not_expanded read-tree --prefix=deep/deeper2 -u deepest
 '
 
+test_expect_success 'sparse index is not expanded: blame' '
```

Comment:
I know this isn't part of the changes you made, but could you make the `blame with pathspec outside sparse definition` ([link](https://github.com/microsoft/git/blob/vfs-2.33.0/t/t1092-sparse-checkout-compatibility.sh#L489-L498)) more robust? Since it just looks for generic "failure" right now, it doesn't verify that sparse index and non-sparse index provide the _same_ error. If you instead do something like

```
test_expect_success 'blame with pathspec outside sparse definition' '
	init_repos &&
	test_sparse_match test_must_fail git blame folder1/a &&
	test_sparse_match test_must_fail git blame folder2/a &&
	test_sparse_match test_must_fail git blame deep/deeper2/a &&
	test_sparse_match test_must_fail git blame deep/deeper2/deepest/a
'
```

it should fail only when the error messages from sparse index and non-sparse index don't match (while still documenting the behavior in the test).

Reply:
Good call!